### PR TITLE
Add default podSet name via webhook mutation

### PR DESCRIPTION
**Pull request by kerthcet**
Before we will have a default name  for unset podSet name, it's auto generated via kubebuilder. But when we have two name unset podSets, they will both have the  name, it violates the API validation of non-repeated names. So remove the annotation in the API but add the default name via webhook muatation only when we have one podSet and its name is empty

Signed-off-by: kerthcet <kerthcet@gmail.com>

#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

**Comment by alculquicondor** [comment]
Follow up to #318

File: apis/kueue/webhooks/workload_webhook_test.go [hunk 1]
```diff
@@ -58,6 +58,24 @@ func TestWorkloadWebhookDefault(t *testing.T) {
 				},
 			},
 		},
+		"will not add default podSet name if podSets number is greater than 1": {
```
Comment (alculquicondor) [inline apis/kueue/webhooks/workload_webhook_test.go:61]: Isn't this already tested?

Comment (kerthcet) [reply apis/kueue/webhooks/workload_webhook_test.go]: oops, didn't notice that. Removed.

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**